tests/run-make/issue-11908/rmake.rs

```diff
@@ -0,0 +1,35 @@
+// A path which contains the same rlib or dylib in two locations
+// should not cause an assertion panic in the compiler.
+// This test tries to replicate the linked issue and checks
+// if the bugged error makes a resurgence.
+
+// Note that this relies on `liburl` to be in the path somewhere else.
+// Our aux-built libraries will collide with liburl (they have
+// the same version listed)
+
+// See https://github.com/rust-lang/rust/issues/11908
+
+//@ ignore-cross-compile
+
+use run_make_support::{dynamic_lib, rustc, tmp_dir};
+use std::fs;
+
+fn main() {
+    let mut tmp_dir_other = tmp_dir();
+    tmp_dir_other.push("other");
+
+    fs::create_dir(tmp_dir_other.clone());
+    rustc().input("foo.rs").crate_type("dylib").codegen_option("prefer-dynamic").run();
+    fs::rename(dynamic_lib("foo"), tmp_dir_other.clone());
+    rustc().input("foo.rs").crate_type("dylib").codegen_option("prefer-dynamic").run();
+    rustc().input("bar.rs").library_search_path(tmp_dir_other.clone()).run();
+    fs::remove_dir_all(tmp_dir());
+
+    fs::create_dir_all(tmp_dir_other.clone());
+    rustc().input("foo.rs").crate_type("rlib").run();
+    let mut tmp_dir_libfoo = tmp_dir();
+    tmp_dir_libfoo.push("libfoo.rlib");
```

Comment:
We should have a helper to construct the name for a rlib (even though it's going to always be `lib{name}.rlib` because *we* control the name generation).